### PR TITLE
[SPARK-13967] [PYSPARK][ML] Added binary Param to Python CountVectorizer 

### PR DESCRIPTION
**Pull request by BryanCutler**
Added binary toggle param to CountVectorizer feature transformer in PySpark.

Created a unit test for using CountVectorizer with the binary toggle on.

### CODE REVIEW

**Comment by BryanCutler** [comment]
cc @MLnick

**Comment by holdenk** [comment]
Since we seem to be adding the same param to multiple models, would it maybe make sense to make this a shared param?

**Comment by BryanCutler** [comment]
> Since we seem to be adding the same param to multiple models, would it maybe make sense to make this a shared param?

Maybe, I'm not sure if there are other uses besides `HashingTF` but I know @MLnick was proposing a more general feature hasher that seems like it would reduce a lot more code duplication..

**Comment by MLnick** [comment]
Will take a look

**Comment by MLnick** [comment]
@holdenk @BryanCutler I'd say we could make `binary` shared, but the only thing is currently the doc is a bit different between them (the doc for `CountVectorizer` mentions the binarization occurs after `minTF` is applied, while `minTF` doesn't exist for `HashingTF`). So if there's a way to easily make them shared but have different doc, then go ahead.

We could later add `minTF` to `HashingTF` I guess, in which case it can definitely be a shared param.

**Comment by MLnick** [comment]
LGTM otherwise.

**Comment by jkbradley** [comment]
+1 for not sharing the Param if the docs (and semantics) differ

File: python/pyspark/ml/feature.py [hunk 1]
```diff
@@ -256,24 +256,33 @@ class CountVectorizer(JavaEstimator, HasInputCol, HasOutputCol, JavaMLReadable,
...
         self.setParams(**kwargs)
 
     @keyword_only
     @since("1.6.0")
-    def setParams(self, minTF=1.0, minDF=1.0, vocabSize=1 << 18, inputCol=None, outputCol=None):
+    def setParams(self, minTF=1.0, minDF=1.0, vocabSize=1 << 18, binary=False, inputCol=None,
+                  outputCol=None):
         """
-        setParams(self, minTF=1.0, minDF=1.0, vocabSize=1 << 18, inputCol=None, outputCol=None)
+        setParams(self, minTF=1.0, minDF=1.0, vocabSize=1 << 18, binary=False, inputCol=None,
```
Comment (mengxr) [inline python/pyspark/ml/feature.py]: Need a backslash at the end. Please check the generated HTML doc.

**Comment by BryanCutler** [comment]
>  Need a backslash at the end. Please check the generated HTML doc

This should have cause the doc checks to fail right?

**Comment by holdenk** [comment]
@BryanCutler it seems that (due to a bug introduced during code review) right now we only hault on doc build errors, not warnings as originally intended. Thanks for noticing this I'll make a follow up JIRA to deal with this.

**Comment by BryanCutler** [comment]
It looks like the option to treat warnings as errors is there, it just gets overwritten in the makefile

**Comment by holdenk** [comment]
Yup I've got a fix but going to cleanup the warnings that got in the meantime too.

**Comment by MLnick** [comment]
@mengxr @jkbradley anything further?

**Comment by jkbradley** [comment]
LGTM, feel free to merge, thanks!

**Comment by MLnick** [comment]
Merged to master. Thanks!

**Pull request closed without merging**